client/src/container/HomePage/index.js

```diff
@@ -1,2 +1,147 @@
-const HomePage = () => {};
+import React, { useState, useEffect } from 'react';
+import propTypes from 'prop-types';
+import {
+  notification,
+  Button,
+  Spin,
+  Rate,
+  TreeSelect,
+  Input,
+  Empty,
+} from 'antd';
+import { AudioOutlined } from '@ant-design/icons';
+
+import axios from 'axios';
+import categories from '../../assets/categories';
+import './style.css';
+
+const { Search } = Input;
+const suffix = (
+  <AudioOutlined
+    style={{
+      fontSize: 16,
+      color: '#1890ff',
+    }}
+  />
+);
+
+const HomePage = ({ history }) => {
+  const [loading, setLoading] = useState(true);
+  const [topcourses, setTopCourses] = useState([]);
+  const [searchCourseName, setSearchCourseName] = useState('');
+  const [cat, setCat] = useState('');
+
+  const fetchCoursesByNameAndCatId = async (catId, courseName) => {
+    try {
+      const { data } = await axios.post(`/api/v1/catId/courseName`, {
+        catId,
+        courseName,
+      });
+      setTopCourses(data);
+    } catch (err) {
+      let error;
+      if (err.response) error = err.response.data.msg || 'Something went wrong';
+      else error = 'No Course with this name';
```

Comment:
**useless error handling here, 
this type of handling shouldn't be handled here, it should be handled in the back or within render.**